src/tool/cpp/cppwinrt/main.cpp

```diff
@@ -13,37 +14,105 @@ namespace xlang
+        { "pch", 0, 1, "<name>", "Specify name of precompiled header file (defaults to pch.h)" },
+        { "include", 0, cmd::option::no_max, "<prefix>", "One or more prefixes to include in input" },
+        { "exclude", 0, cmd::option::no_max, "<prefix>", "One or more prefixes to exclude from input" },
+        { "base", 0, 0, {}, "Generate base.h unconditionally" },
+        { "opt", 0, 0, {}, "Generate component projection with unified construction support" },
+        { "help", 0, cmd::option::no_max, {}, "Show detailed help with examples" },
+        { "root", 0, 1 }, // "Root folder name for projection headers (defaults to winrt)
+        { "filter" }, // One or more prefixes to include in input (same as -include)
+        { "lib", 0, 1 }, // Specify library prefix (defaults to winrt)
+        { "license", 0, 0 }, // Generate license comment
+        { "brackets", 0, 0 }, // Use angle brackets for #includes (defaults to quotes)
+    };
+
+    static void printUsage(writer& w, bool details)
     {
-        std::vector<cmd::option> options
+        static auto printColumns = [](writer& w, std::string_view const& col1, std::string_view const& col2)
+        {
+            w.write_printf("  %-20s%s\n", col1.data(), col2.data());
+        };
+
+        static auto printOption = [](writer& w, cmd::option const& opt)
         {
-            { "input", 0 },
-            { "reference", 0 },
-            { "output", 0, 1 },
-            { "component", 0, 1 },
-            { "filter", 0 },
-            { "name", 0, 1 },
-            { "verbose", 0, 0 },
-            { "overwrite", 0, 0 },
-            { "prefix", 0, 0 },
-            { "license", 0, 0 },
-            { "pch", 0, 1 },
-            { "include", 0 },
-            { "exclude", 0 },
-            { "root", 0, 1 },
-            { "base", 0, 0 },
-            { "lib", 0, 1 },
-            { "opt", 0, 0 },
-            { "bracket", 0, 0 },
+            if(opt.desc.empty())
+            {
+                return;
+            }
+            printColumns(w, w.write_temp("-% %", opt.name, opt.arg), opt.desc);
         };
 
+        auto format = R"(
+C++/WinRT v%
+Copyright (c) % Microsoft Corporation. All rights reserved.
+
+  cppwinrt.exe [options...]
+
+Options:
+
+%  ^@<path>             Response file containing command line options
+
+Where <spec> is one or more of:
+
+  path                Path to winmd file or recursively scanned folder
+  local               Local ^%WinDir^%\System32\WinMetadata folder
+  sdk[+]              Current version of Windows SDK [with extensions]
+  10.0.12345.0[+]     Specific version of Windows SDK [with extensions]
+)";
+        time_t t{ time(nullptr) };
+        tm tm{};
+        localtime_s(&tm, &t);
+
+        w.write(format, 
+            XLANG_VERSION_STRING, 
+            1900 + tm.tm_year, 
+            bind_each(printOption, options));
+
+        if (details)
```

Comment:
I don't think any of this needs to be in the tool. We can point to the docs page. Beyond that it seems impractical to try to make the tool self-documenting without substantially more effort. I think a list of options is sufficient and perhaps a link to the docs. Then we can focus on having quality detailed usage information in one place (online).